x/staking/src/abci_handler.rs

```diff
@@ -68,11 +75,19 @@ impl<
 
     fn end_block<DB: Database, CTX: TransactionalContext<DB, SK>>(
         &self,
-        _ctx: &mut CTX,
+        ctx: &mut CTX,
         _request: RequestEndBlock,
     ) -> Vec<ValidatorUpdate> {
-        vec![]
-        // self.keeper.end_block(ctx, request)
+        match self.keeper.block_validator_updates(ctx) {
+            Ok(updates) => updates,
+            Err(err) => panic!(
+                "Error thrown in method 'end_block'. It can broke blockchain.
+                Original error: {:?}",
+                err
```

Comment:
Similar to above, the keeper `block_validator_updates` method should be infallible.